utils/S3UtilsMongoClient.js

```diff
@@ -765,6 +784,67 @@ class S3UtilsMongoClient extends MongoClientInterface {
             });
         });
 
+        // parse all location and reflect the data in the bucket
+        Object.keys((res.bucket || {})).forEach(bucket => {
+            if (!dataMetrics.bucket[bucket].locations) {
```

Comment:
This also assumes dataMetrics.bucket[bucket] is always defined.

Reply:
Yes, but here we also rely on the dataMetrics that we computed from `res` . We are missing types here (TS) to make it clearer... Anyway the same approach is used since years for the accounts, without any problem.